dashboard/test/controllers/level_starter_assets_controller_test.rb

```diff
@@ -54,6 +70,72 @@ class LevelStarterAssetsControllerTest < ActionController::TestCase
     assert_equal 'image/png', response.headers['Content-Type']
     assert_equal 'inline', response.headers['Content-Disposition']
   end
+
+  test 'upload: forbidden for non-levelbuilders' do
+    sign_in create(:student)
+    post :upload, params: {level_name: create(:level).name, files: []}
+    assert_response :forbidden
+  end
+
+  test 'upload: forbidden if not in levelbuilder_mode' do
+    Rails.application.config.stubs(:levelbuilder_mode).returns(false)
+    sign_in create(:levelbuilder)
+    post :upload, params: {level_name: create(:level).name, files: []}
+    assert_response :forbidden
+  end
+
+  test 'upload: raises an error if 2+ files are uploaded' do
+    sign_in create(:levelbuilder)
+
+    e = assert_raises do
+      post :upload, params: {level_name: create(:level).name, files: ['file-1', 'file-2']}
+    end
+    assert_equal 'One file upload expected. Actual: 2', e.message
+  end
+
+  test 'upload: returns unprocessable_entity if file fails to upload' do
+    LevelStarterAssetsController.any_instance.
+      expects(:get_object).
+      returns(@file_obj)
+    @file_obj.expects(:upload_file).returns(false)
+
+    sign_in create(:levelbuilder)
+    post :upload, params: {level_name: create(:level).name, files: [@file]}
+
+    assert_response :unprocessable_entity
+  end
+
+  test 'upload: returns unprocessable_entity if file uploads but starter asset is not added' do
+    LevelStarterAssetsController.any_instance.
+      expects(:get_object).
+      returns(@file_obj)
+    @file_obj.expects(:upload_file).returns(true)
+    Level.any_instance.expects(:save).returns(false)
+
+    sign_in create(:levelbuilder)
+    post :upload, params: {level_name: create(:level).name, files: [@file]}
+
+    assert_response :unprocessable_entity
+  end
+
+  test 'upload: returns summary if file uploads and starter asset is added' do
+    LevelStarterAssetsController.any_instance.
+      expects(:get_object).
+      returns(@file_obj)
+    @file_obj.expects(:upload_file).returns(true)
+
+    sign_in create(:levelbuilder)
+    level = create :level
+    post :upload, params: {level_name: level.name, files: [@file]}
+
+    level.reload
+    assert_equal 1, level.starter_assets.length
+    assert_response :success
+    summary = JSON.parse(response.body)
+    assert_equal @filename, summary['filename']
+    assert_equal 'image', summary['category']
+    assert_equal 123, summary['size']
+  end
```

Comment:
great tests! please add one test case for a friendly name containing at least the `'` and `"` characters.